scio-smb/src/test/scala/com/spotify/scio/smb/util/SMBMultiJoinTest.scala

```diff
@@ -0,0 +1,129 @@
+package com.spotify.scio.smb.util
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.avro._
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.smb.toSortMergeBucketSCollection
+import com.spotify.scio.testing.PipelineSpec
+import com.spotify.scio.util.MultiJoin
+import com.spotify.scio.values.SCollection
+import org.apache.avro.Schema
+import org.apache.avro.Schema.Field
+import org.apache.avro.generic.{GenericData, GenericRecord}
+import org.apache.beam.sdk.extensions.smb.{AvroSortedBucketIO, SortedBucketIO, TargetParallelism}
+import org.apache.beam.sdk.values.TupleTag
+import org.apache.commons.io.FileUtils
+import org.scalatest.Assertion
+
+import java.nio.file.{Files, Path}
+import scala.jdk.CollectionConverters._
+import scala.util.Random
+
+class SMBMultiJoinTest extends PipelineSpec {
+
+  private val schema = Schema.createRecord(
+    "User",
+    "",
+    "com.spotify.scio.smb",
+    false,
+    List(
+      new Field("key", Schema.create(Schema.Type.INT), "", -1),
+      new Field("value", Schema.create(Schema.Type.STRING), "", "")
+    ).asJava
+  )
+
+  implicit private val coder: Coder[GenericRecord] = Coder.avroGenericRecordCoder(schema)
+
+  private val keyFn: GenericRecord => Integer = _.get("key").toString.toInt
+
+  "SMBMultiJoin" should "have parity with a 5-way CoGroup" in withNumSources(5) { inputs =>
```

Comment:
I think it would be simpler to just move this test case into `SortMergeBucketParityIT` :)